aspnetcore/blazor/fundamentals/render-modes.md

```diff
@@ -0,0 +1,295 @@
+---
+title: ASP.NET Core Blazor render modes
+author: guardrex
+description: Learn about Blazor render modes and how to apply them in Blazor Web Apps.
+monikerRange: '>= aspnetcore-8.0'
+ms.author: riande
+ms.custom: mvc
+ms.date: 09/08/2023
+uid: blazor/fundamentals/render-modes
+---
+# ASP.NET Core Blazor render modes
+
+This article explains how to establish hosting models for rendered Razor components in Blazor Web Apps, either at compile time or runtime.
+
+## Render modes
+
+Every component in a Blazor Web App adopts a *render mode* to determine the hosting model that it uses, where it's rendered, and whether or not it's interactive with C# code.
+
+The following table shows the four potential render mode scenarios for rendering Razor components in a Blazor Web App. The render mode attributes in the *Render mode attribute* column are applied to components with the [`@attribute` Razor directive](xref:mvc/views/razor#attribute). Later in this article, examples are shown for each render mode scenario.
```

Comment:
The Server, WebAssembly, and Auto render modes also have prerendered variants.

Reply:
Also, just a heads up, that in RC2 we will get Razor compiler support for the new `@rendermode` directive: https://github.com/dotnet/razor/issues/9068.

Reply:
That will ... ***should be*** 🤞🍀 ... an easy find-'n-replace gesture for me when the time comes.

Reply:
Prerendering is enabled by default for all render models in .NET 8. To turn it off you do this: `@attribute [RenderModeServer(prerender: false)]` or `@rendermode="new ServerRenderMode(prerender: false)"`

Reply:
Ah ... yeah. I knew about that, but this whole doc is whatever was on my mind while typing. I'll add that in the morning.